backend/src/main/java/com/capstone/moneytree/validator/Validator.java

```diff
@@ -0,0 +1,15 @@
+package com.capstone.moneytree.validator;
+
+
+/**
+ * The purpose of this class is to provide a common interface for all validator
+ * classes for our entities.
+ */
+public interface Validator {
+
+   boolean supports(Class<?> var1);
+
+   void validate(Object var1);
+
+   boolean isValid(Object var1);
+}
```

Comment:
We could use the [spring validator interface](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/validation/Validator.html)

Reply:
Thks for the link, I actually tried to do so before implementing our own interface which looks super simialr to the spring one. The reason why I settle by providing our own interface is that in the spring interface, within the validate method, we need to provide an error object. But the way we design our rest api is that we do no deal with errors, we deal with exception and have our exception handlers return the response to the user.

That is why we have our own interface, only difference is that the validate() methods takes in an object to validate with no error object as opposed to spring.